px4_ros2_cpp/include/px4_ros2/control/setpoint_types/fixedwing/fw_control_setpoint.hpp

```diff
@@ -0,0 +1,62 @@
+/****************************************************************************
+ * Copyright (c) 2023 PX4 Development Team.
+ * SPDX-License-Identifier: BSD-3-Clause
+ ****************************************************************************/
+
+#pragma once
+
+#include <px4_msgs/msg/fw_lateral_control_setpoint.hpp>
+#include <px4_msgs/msg/fw_longitudinal_control_setpoint.hpp>
+#include <Eigen/Eigen>
+#include <optional>
+
+#include <px4_ros2/common/setpoint_base.hpp>
+#include <px4_ros2/utils/geodesic.hpp>
+
+namespace px4_ros2
+{
+/** \ingroup setpoint_types
+ *  @{
+ */
+
+/**
+ * @brief Setpoint type for smooth position and heading control
+*/
+    class FwControlSetpointType : public SetpointBase
+    {
+    public:
+	explicit FwControlSetpointType(Context & context);
+
+	~FwControlSetpointType() override = default;
+
+	Configuration getConfiguration() override;
+
+	/**
+	 * @brief Update
+	 *
+	 * Unset optional values are not controlled
+	 *
+	 */
+	void update(
+			    const float course_setpoint,
+				const float heading_setpoint, 
```

Comment:
@mahimayoga Let's call this `airspeed_reference_direction` to be very precise. I also named it that way in PX4.